test/test_distributions.py

```diff
@@ -704,18 +717,45 @@ def _check_sample(funsor_dist, sample_inputs, inputs, atol=1e-2, rtol=None,
         else:
             raise ValueError("invalid test statistic")
 
-        grad_targets = [v.data for v in list(funsor_dist.params.values())[:-1]]
-        actual_grads = torch.autograd.grad(actual_stat.reduce(ops.add).sum().data, grad_targets, allow_unused=True)
-        expected_grads = torch.autograd.grad(
-            expected_stat.reduce(ops.add).sum().data, grad_targets, allow_unused=True)
+        diff = (actual_stat - expected_stat).reduce(ops.add).data
+        return diff.sum(), diff
 
-        assert_close(actual_stat, expected_stat, atol=atol, rtol=rtol)
 
-        for actual_grad, expected_grad in zip(actual_grads, expected_grads):
-            if expected_grad is not None:
-                assert_close(actual_grad, expected_grad, atol=atol, rtol=rtol)
+def _check_sample(funsor_dist_class, params, sample_inputs, inputs, atol=1e-2,
```

Comment:
This is rewritten to also test for JAX distributions. The functionality should be the same as before.